subtask_9.3/src/DataBuffer.java

```diff
@@ -0,0 +1,46 @@
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class DataBuffer {
+
+    private Queue<Integer> buffer;
```

Comment:
Same here. Low level thread management. Try to improve this by using concurrent package instruments. 
ArrayBlockingQueue, Locks, etc